deepchem/models/tensorgraph/models/symmetry_function_regression.py

```diff
@@ -285,26 +292,50 @@ def minimize_structure(self, X, atomic_nums, constraints=None):
     num_atoms = atomic_nums.shape[0]
 
     res = scipy.optimize.minimize(
-        self.pred_one,
-        X,
-        args=(atomic_nums, constraints),
-        jac=self.grad_one,
-        method="BFGS",
-        tol=1e-6,
-        options={'disp': True})
+      self.pred_one,
+      X,
+      args=(atomic_nums, constraints),
+      jac=self.grad_one,
+      method="BFGS",
+      tol=1e-6,
+      options={'disp': True})
 
     return res.x.reshape((num_atoms, 3))
 
   def build_graph(self):
 
-    self.atom_numbers = Feature(shape=(None, self.max_atoms), dtype=tf.int32)
-    self.atom_flags = Feature(shape=(None, self.max_atoms, self.max_atoms))
-    self.atom_feats = Feature(shape=(None, self.max_atoms, 4))
+    self.mode = Feature(dtype=tf.bool) # if true then we need to featurize
+    self.dequeue_object = Feature() # either atom_feats or featurized
+
+    def true_fn():
+      r_feat = Reshape(
+        shape=[None, self.max_atoms, 4],
+        in_layers=[self.dequeue_object])
+      r_feat.create_tensor()
+
+      ani_feat = ANIFeat(
+        in_layers=[r_feat],
+        max_atoms=self.max_atoms)
+
+      ani_feat.create_tensor()
+      return ani_feat
 
-    previous_layer = ANIFeat(
-        in_layers=self.atom_feats, max_atoms=self.max_atoms)
+    def false_fn():
 
-    self.featurized = previous_layer
+      # dequeue yields a sparse tensor for the entire shard
+
+      r_feat = Reshape(
+        shape=[None, self.max_atoms, 769],
```

Comment:
Note to self, un hardcode the 769 (this is for 768+1 computed for CNOH). This is a fixed value and should be pre-computed to avoid calling tf.shape.

Reply:
Could you add a comment here explaining the 769 and adding a TODO?